comp/dogstatsd/server/server_uds_test.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build !windows
+
+// UDS won't work in windows
+
+package server
+
+import (
+	"net"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/DataDog/datadog-agent/comp/dogstatsd/listeners"
+)
+
+func TestUDSReceiver(t *testing.T) {
+	socketPath := filepath.Join(t.TempDir(), "dsd.socket")
+
+	cfg := make(map[string]interface{})
+	cfg["dogstatsd_port"] = listeners.RandomPortName
+	cfg["dogstatsd_no_aggregation_pipeline"] = true // another test may have turned it off
+	cfg["dogstatsd_socket"] = socketPath
+
+	deps := fulfillDepsWithConfigOverride(t, cfg)
+	demux := deps.Demultiplexer
+
+	conn, err := net.Dial("unixgram", socketPath)
+	require.NoError(t, err, "cannot connect to DSD socket")
+	defer conn.Close()
+
+	testReceive(t, conn, demux)
+}
+
+func TestUDSReceiverDisabled(t *testing.T) {
+	cfg := make(map[string]interface{})
+	cfg["dogstatsd_port"] = listeners.RandomPortName
+	cfg["dogstatsd_no_aggregation_pipeline"] = true // another test may have turned it off
+	cfg["dogstatsd_socket"] = ""                    // disabled
+
+	_ = fulfillDepsWithConfigOverride(t, cfg)
+
+	socketPath := filepath.Join(t.TempDir(), "dsd.socket")
+	_, err := net.Dial("unixgram", socketPath)
```

Comment:
I may be missing something, this test seems a bit non-sensical. TempDir() creates a new randomly named directory each time it is called, so the name is not known, nor the directory exists when we configure and start the dogstatsd server. Even if the socket is not disabled as expected, it won't be at this random location.

Reply:
Yeah true. I will just remove this altogether, and rely on the other test that will also stop the server and ensure the conn fails if called after that.

Reply:
Updated. I actually left the test to check for 
```go
require.False(t, deps.Server.UdsListenerRunning())
```